invenio_db/utils.py

```diff
@@ -129,3 +132,8 @@ def has_table(engine, table):
     except AttributeError:
         # SQLAlchemy <1.4
         return engine.has_table(table)
+
+
+def now():
+    """Creates a datetime object."""
+    return datetime.now(timezone.utc).replace(tzinfo=None)
```

Comment:
IMHO I would prefer that we find a more "SQLAlchemy way" to avoid this footgun:

- It's going to be difficult to teach newcomers and (unfortunately, manually) enforce thoughout the codebase doing `from invenio_db import now` everytime we store UTC/non-TZ-aware datetimes in the DB.
- It's going to be difficult to catch datetime-related bugs in tests, since often [we just "ignore" the values of  `created`/`updated` fields](https://github.com/inveniosoftware/invenio-rdm-records/blob/7e17fad16868ed40c2b65082246dcac77ad45fb1/tests/records/test_records_communities.py#L56) in responses and focus more on "data content" (which is not a bad thing when applied consiously).
- I would be more in favour towards finding some config/setting/param in SQLAlchemy datetime columns or even defining a new custom DateTime column type that "does the right thing"™️ on (de)serialization which we can then gradually replace throughout the codebase in models that:
  - Doesn't affect Alembic migrations (e.g. column types in Postgres are still `TIMESTAMP WITHOUT TIME ZONE`)
  - Makes sure that Python `datetime` objects with the UTC timezone, get transformed accordingly
  - Explicitly complains/errors when non-UTC timezone-aware `datetime`s are passed, so that it's not the humans/reviewers that have to check, but the code does it instead.

---

On a side note, I think we've never explicitly wrote this anywhere, but the common practice throughout Invenio is to store/persist UTC times and convert to/from local timezone-aware times in the UI. I don't know where to log this "decision", but [this post](https://simonwillison.net/2024/Nov/27/storing-times-for-human-events/) is a good supplementary read, from which I get the feeling that for us storing UTC is better.